core/modules/user/user.admin.inc

```diff
@@ -399,14 +399,50 @@ function user_login_settings($form, &$form_state) {
     ),
   );
 
+  $form['user_login_appearance_setting'] = array(
+    '#type' => 'fieldset',
+    '#title' => '',
+  );
+  $user_login_page_tab_options = array(
+    'tabs' => t('Use tabs'),
+    'links' => t('Use links'),
+    'simplified' => t('Show simplified page'),
+  );
+  $user_login_page_default_settings = $core_config->get('user_login_appearance_setting');
+  $form['user_login_appearance_setting']['user_login_appearance_setting'] = array(
+    '#title' => t('Login page options'),
+    '#type' => 'radios',
+    '#options' => $user_login_page_tab_options,
+    '#default_value' => $user_login_page_default_settings ? $user_login_page_default_settings : 'tabs',
+    '#config' => 'system.core',
+    'tabs' => array(
+      '#description' => t('Show tabs for "Create new account", "Log in", and "Reset password" links.'),
+    ),
+    'links' => array(
+      '#description' => t('Show plain links to these pages.'),
+    ),
+    'simplified' => array(
+      '#description' => t('Show plain links to these pages, and hide all layout regions, such as headers, footers, and sidebars.'),
+    ),
+  );
   $form['flood_log_failed_attempts'] = array(
     '#type' => 'checkbox',
     '#title' => t('Log excessive failed login attempts.'),
     '#default_value' => $flood_config->get('flood_log_failed_attempts'),
     '#description' => t('If there are excessive failed login attempts, the offending IP address or user account will be temporarily blocked. By default these events will be logged, as it helps identify brute force login attacks. You may want to disable logging them if for example, you are using the dblog module, and want to avoid database writes.'),
   );
 
-  return system_settings_form($form);
+  $form = system_settings_form($form);
+  // Add another submit handler to clear caches.
+  $form['#submit'][] = 'user_login_settings_submit';
+  return $form;
+}
+
+/**
+ * Second Form submission handler for user_login_settings().
+ */
+function user_login_settings_submit($form, &$form_state) {
+  cache_clear_all();
```

Comment:
cache_clear_all() doesn't do the trick for me.

I tried backdrop_flush_all_caches() instead and that does seem to work, but it really does too much (we don't care about the entity cache...).

Function `menu_rebuild()` alone does the trick for me, but it still seems to do a little too much.